### ISSUE TITLE
Tag is not extracted from packed-refs

### ISSUE THREAD

**Issue by janmisek** [rwjblue/git-repo-info#17]
Tag is not extracted from  `packed-refs` even `git describe` shows proper tag.

I traced down problem to incompatiblity between my `packed-refs` file and `findPackedTag` method logic.  

My packed-refs file ends with following fragment:

```
67d2a502b55a8b9fc5983b8339f26fba6d288878 refs/tags/1.2.5
486ea0927e8eff57c5b688c36058ad0cbd84890d refs/tags/1.2.6
^e801890a91d492f258c80298882d108ac93e746e
bb70a4667c7a8a0cc3f7a0838d5d116ba9d47731 refs/tags/1.2.7
^9ecb16b2b2bd39e23a5a6d76ac0a3a75bb071a48
c2b21d5081e311ca1de412fa7346955e64a7de23 refs/tags/1.2.8
^f978fe49797023d9bd47068b39d3a477e51b1bbf
```

There are shas prefixed with `^`. These prefixed shas are actually looked up (passed to `findPackedTag` method ) but these are on another line then tag itself.

e.g:
`findPackedTag` got sha f978fe49797023d9bd47068b39d3a477e51b1bbf
but tag is actually referenced by c2b21d5081e311ca1de412fa7346955e64a7de23 refs/tags/1.2.8 

then following filter inside the `findPackedTag` returns nothing:

```
 var tagLine = packedRefsFile.split('\n').filter(function(line) {
      return line.indexOf('refs/tags') > -1 && line.indexOf(sha) > -1;
    })[0];
```

This is above my knowledge of git. Please could you advise. My versioning of ember-cli app stopped working.

Environment:
git-repo-info: 1.0.4 but even 1.1.2 behaves same
git 2.5.0

**Comment by seawatts** [comment]
I just talked with Circle CI and they responded with this:

```
Hi Chris,

What you're seeing is packed refs, where all the ref info is kept in a single file - https://www.kernel.org/pub/software/scm/git/docs/git-pack-refs.html
cat .git/packed-refs should show all the info you expect.

It's something that git itself does, you don't normally notice because git will unpack refs for branches whose heads you move by committing/merging etc.

There's no "easy" way to unpack packed refs although http://stackoverflow.com/a/3333428 has a suggestion which I haven't tried.

-Gordon
```

**Comment by seawatts** [comment]
I haven't tried that script out yet but I don't really want to. There has to be a better solution. This use to work like 3 weeks ago.

**Comment by rwjblue** [comment]
https://github.com/rwjblue/git-repo-info/pull/18 should have fixed, and is released in 1.1.3.
